sdks/java/core/src/main/java/org/apache/beam/sdk/values/WindowedValues.java

```diff
@@ -61,6 +66,172 @@
 public class WindowedValues {
   private WindowedValues() {} // non-instantiable utility class
 
+  public static <T> Builder<T> builder() {
+    return new Builder<>();
+  }
+
+  /** Create a Builder that takes element metadata from the provideed delegate. */
+  public static <T> Builder<T> builder(WindowedValue<T> template) {
+    return new Builder<T>()
```

Comment:
thi is great, so if we start to expand with new fields, this is the place to propagate them.